todo-list-ui/src/app/todo-edit-modal/todo-edit-modal.component.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Component,
+  EventEmitter,
+  inject,
+  Input,
+  Output,
+  TemplateRef,
+} from '@angular/core';
+import { FormControl, FormsModule, ReactiveFormsModule } from '@angular/forms';
+import {
+  NgbDatepickerModule,
+  NgbModal,
+  NgbModalRef,
+} from '@ng-bootstrap/ng-bootstrap';
+
+@Component({
+  selector: 'app-todo-edit-modal',
+  standalone: true,
+  imports: [ReactiveFormsModule, FormsModule, NgbDatepickerModule],
+  templateUrl: './todo-edit-modal.component.html',
+})
+export class TodoEditModalComponent {
+  @Input({ required: true }) id: string | undefined;
+  @Input({ required: true }) title: string | undefined;
+  @Output() newTitle = new EventEmitter<string>();
+
+  private modalService = inject(NgbModal);
+  private modalInstance?: NgbModalRef;
+  inputTodo = new FormControl('', { nonNullable: true });
+
+  open(content: TemplateRef<any>): Promise<unknown> {
+    this.inputTodo.setValue(this.title ?? '');
+    this.modalInstance = this.modalService.open(content);
+    return this.modalInstance.result;
+  }
+
+  editTodo() {
+    this.newTitle.emit(this.inputTodo.value);
+    this.modalInstance?.close();
+  }
+
+  cancel() {
+    this.newTitle.emit('');
```

Comment:
Will this set the title to empty string (`''`) if the edit-modal is cancelled ?

Also, Outputs are akin to events, so usually you want to word them in past (when the event already happened, e.g. `titleUpdated`, `submitted` -- since the form was 'sent')